Lib/test/test_remote_pdb.py

```diff
@@ -78,6 +79,697 @@ def get_output(self) -> List[dict]:
+            ("user", {"prompt": "...   ", "input": "b ]"}),
+        ]
+        self.do_test(
+            incoming=incoming,
+            expected_outgoing=[
+                {"reply": "blah [\nblah ]"},
+                {"reply": ""},
+                {"reply": "b ["},
+                {"reply": "!b [\nb ]"},
+            ],
+            expected_state={"state": "pdb"},
+        )
+
+    def test_handling_interact_prompts(self):
+        """Test responding to pdb's interact mode prompts."""
+        incoming = [
+            ("server", {"command_list": ["b"]}),
+            ("server", {"prompt": ">>> ", "state": "interact"}),
+            ("user", {"prompt": ">>> ", "input": "blah ["}),
+            ("user", {"prompt": "... ", "input": "blah ]"}),
+            ("server", {"prompt": ">>> ", "state": "interact"}),
+            ("user", {"prompt": ">>> ", "input": ""}),
+            ("server", {"prompt": ">>> ", "state": "interact"}),
+            ("user", {"prompt": ">>> ", "input": "b ["}),
+            ("user", {"prompt": "... ", "input": "b ]"}),
+        ]
+        self.do_test(
+            incoming=incoming,
+            expected_outgoing=[
+                {"reply": "blah [\nblah ]"},
+                {"reply": ""},
+                {"reply": "b [\nb ]"},
+            ],
+            expected_state={"state": "interact"},
+        )
+
+    def test_retry_pdb_prompt_on_syntax_error(self):
+        """Test re-prompting after a SyntaxError in a Python expression."""
+        incoming = [
+            ("server", {"prompt": "(Pdb) ", "state": "pdb"}),
+            ("user", {"prompt": "(Pdb) ", "input": " blah ["}),
+            ("user", {"prompt": "(Pdb) ", "input": "blah ["}),
+            ("user", {"prompt": "...   ", "input": " blah ]"}),
+        ]
+        self.do_test(
+            incoming=incoming,
+            expected_outgoing=[
+                {"reply": "blah [\n blah ]"},
+            ],
+            expected_stdout_substring="*** IndentationError",
+            expected_state={"state": "pdb"},
+        )
+
+    def test_retry_interact_prompt_on_syntax_error(self):
+        """Test re-prompting after a SyntaxError in a Python expression."""
+        incoming = [
+            ("server", {"prompt": ">>> ", "state": "interact"}),
+            ("user", {"prompt": ">>> ", "input": "!blah ["}),
+            ("user", {"prompt": ">>> ", "input": "blah ["}),
+            ("user", {"prompt": "... ", "input": " blah ]"}),
+        ]
+        self.do_test(
+            incoming=incoming,
+            expected_outgoing=[
+                {"reply": "blah [\n blah ]"},
+            ],
+            expected_stdout_substring="*** SyntaxError",
+            expected_state={"state": "interact"},
+        )
+
+    def test_handling_unrecognized_prompt_type(self):
+        """Test fallback to "dumb" single-line mode for unknown states."""
+        incoming = [
+            ("server", {"prompt": "$ ", "state": "shell"}),
```

Comment:
I understand the `"state"` could be anything, including `"shell"`, but here `"shell"` is a bit distracting because people will wonder if that actually has something to do with shell. Could we use something that `pdb` actually uses here? `confirm` or `commands` could both work.